evm_arithmetization/src/cpu/kernel/asm/core/call.asm

```diff
@@ -396,6 +397,14 @@ call_too_deep:
     // (Old context) stack: new_ctx
 %endmacro
 
+%macro set_ctx_pruning_flag
+    // stack: old_ctx
+    DUP1 
+    // Check whether we are going to a context such that cur_ctx > new_ctx
+    GET_CONTEXT SWAP1 LT
```

Comment:
```suggestion
    // Check whether we are going to a context such that cur_ctx > new_ctx
    GET_CONTEXT DUP2 LT
```

Reply:
More generally, you barely (if ever) need this method. For example, during a `CALL`, you know you will never prune context, while during `EXTCODESIZE` you always will.